### PR TITLE
fix: lit encryption issues

### LINKED ISSUE #256: Switching wallet in Metamask fails to reconnect to Lit Protocol properly resulting in error while creating encrypted request
# Problem(s)

Fails to reconnect to Lit Protocol when switching wallet address in Metamask.

Image

# Steps to Reproduce

1. Connect wallet
2. Show Dashboard
3. Enable decrypted requests
4. Change wallets in Metamask
5. Observe that not prompted to reconnect to Lit Protocol with the new wallet
6. If you try to create an encrypted request, you get the error shown in the screenshot above.

# Solution

We need to handle the Lit Protocol connection state better when changing Metamask addresses.

### PR DESCRIPTION
**Pull request by rodrigopavezi**
## Changes

chore: Update dependencies and improve CipherProvider integration in invoice forms

- Updated various dependencies across the project, including @requestnetwork packages to their latest versions.
- Enhanced CipherProvider integration in Create Invoice Form and Invoice Dashboard, ensuring better wallet management and session handling.
- Refactored code to improve type definitions and streamline wallet connection logic.
- Minor adjustments to the create-invoice-form and view-requests components for better functionality and maintainability.

Fixes #256 #257 

## Summary by CodeRabbit

- **New Features**
	- Enhanced wallet management functionality in the invoice form and dashboard components.
	- Introduced a new method for disconnecting wallets.
  
- **Bug Fixes**
	- Addressed issues related to Lit Protocol Encryption across multiple packages.

- **Documentation**
	- Updated changelogs for `@requestnetwork/create-invoice-form`, `@requestnetwork/invoice-dashboard`, and `@requestnetwork/payment-widget` to reflect recent changes and version updates.

- **Chores**
	- Incremented version numbers and updated dependencies in the `package.json` files for relevant packages.

### CODE REVIEW

File: packages/create-invoice-form/src/lib/create-invoice-form.svelte [hunk 1]
```diff
@@ -25,11 +25,15 @@
...
+  interface CipherProvider extends CipherProviderTypes.ICipherProvider {
+    disconnectWallet: () => void;
+  }
+
   export let config: IConfig;
   export let wagmiConfig: WagmiConfig;
   export let requestNetwork: RequestNetwork | null | undefined;
   export let currencies: CurrencyTypes.CurrencyInput[] = [];
-  let cipherProvider: CipherProviderTypes.ICipherProvider | undefined;
+  let cipherProvider: CipherProvider | undefined;
```
[the remark this answers is not in the document]
Comment (rodrigopavezi) [reply packages/create-invoice-form/src/lib/create-invoice-form.svelte:36]: Yes, that is confirmed

File: packages/invoice-dashboard/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@requestnetwork/invoice-dashboard",
-  "version": "0.11.8",
+  "version": "0.11.9",
```
Comment (MantisClone) [inline packages/invoice-dashboard/package.json:3]: Have there been any features merged since the last version? Should this be a minor version bump?

**Review by MantisClone — COMMENTED** [review]
Partial review. I can take a closer look after the kids are in bed.

**Review by MantisClone — APPROVED** [review]
I have trouble imagining how the app acts based on the logical changes. But I think the logic looks okay. So I'll approve 👍 pending comment resolution.

File: packages/invoice-dashboard/package.json (hunk 2 above)
Comment (rodrigopavezi) [reply packages/invoice-dashboard/package.json:3]: Not really, it only has fixes for what is already there.

**Pull request merged**